### PR TITLE
Add more APIs to work with AutoIdVector.

### PR DESCRIPTION
**Pull request by Ms2ger**
This change is [Reviewable](https://reviewable.io/reviews/servo/rust-mozjs/305)

### CODE REVIEW

**Comment by Ms2ger** [comment]
This is working towards https://github.com/servo/servo/issues/13144

File: src/rust.rs [hunk 1]
```diff
@@ -882,6 +884,48 @@ impl JSNativeWrapper {
...
+impl IdVector {
+    pub fn new(cx: *mut JSContext) -> IdVector {
+        unsafe {
+            let vector = CreateAutoIdVector(cx);
+            assert!(!vector.is_null());
+            IdVector(vector)
+        }
+    }
+
+    pub unsafe fn get(&self) -> *mut AutoIdVector {
```
Comment (nox) [inline src/rust.rs]: This can be safe.

File: src/rust.rs [hunk 2]
```diff
@@ -882,6 +884,48 @@ impl JSNativeWrapper {
...
+        self.0
+    }
+
+    pub fn append(&mut self, id: jsid) -> bool {
```
Comment (nox) [inline src/rust.rs]: This should be unsafe, given we can pass arbitrary `jsid` values.

**Pull request merged**